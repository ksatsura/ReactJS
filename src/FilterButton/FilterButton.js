//imports
import React from 'react';
import '../style.css';

const filterButton = (props) => {
    return (
        <button id={props.id} className={props.styles} onClick={props.onClick}>{props.id}</button>
    );
};

export default filterButton;