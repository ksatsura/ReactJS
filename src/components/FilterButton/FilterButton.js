//imports
import * as React from 'react';
import classNames from 'classnames';
import { connect } from  'react-redux';
import { toggleSearchFilter, toggleSortFilter } from '../../actions/searchActions';
import '../../style.css';

// @flow
type Props = {
    id: string, 
    searchBy: string, 
    sortBy: string, 
    handleFilterClick: (id: string) =>  void, 
    handleSortClick: (id: string) =>  void, 
    name: string,
};

export const FilterButton = (props: Props) => {

    const { id, searchBy, sortBy, handleFilterClick, handleSortClick, name } = props;
    const handleClick = (id === 'title' || id === 'genre') ? handleFilterClick : handleSortClick;
    let styles = (id === 'title' || id === 'genre') 
        ? classNames({ [`search-filter-btn-${id}`]: true }, { focused: (searchBy === id) })
        : classNames('search-results-btn', { selected: (sortBy === id) });

    return (
        <button id={id} className={styles} onClick={(e) => handleClick(e.target.id) }>{name}</button>
    );
    
};

const mapStateToProps = state => {
    return {
        searchBy: state.filmListReducer.searchBy,
        sortBy: state.filmListReducer.sortBy,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        handleFilterClick: (id) => dispatch(toggleSearchFilter(id)),
        handleSortClick: (id) => dispatch(toggleSortFilter(id)),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(FilterButton);