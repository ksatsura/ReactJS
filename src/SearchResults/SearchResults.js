//imports
import React, { Component } from 'react';
import FilterButton from '../FilterButton/FilterButton.js';
import '../style.css';

class SearchResults extends Component {

    constructor (props) {
        super(props);
        this.state = {
            selected: 'rating',
        };
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(event) {
        this.setState({selected: event.target.id});
    }

    render() {
        return (
            <div className='search-results'>
                <p className='results-title'>{this.props.count + ' movies found'}</p>
                <div className='results-filter'> 
                    <p className='results-filter-title'>Sort by</p>
                    <FilterButton id='release date' styles={ this.state.selected === 'release date' ? 'search-results-btn selected' : 'search-results-btn'} handleClick={this.handleClick} />
                    <FilterButton id='rating' styles={ this.state.selected === 'rating' ? 'search-results-btn selected' : 'search-results-btn'} handleClick={this.handleClick} />
                </div>
            </div>
    
        );
    }
}

export default SearchResults;