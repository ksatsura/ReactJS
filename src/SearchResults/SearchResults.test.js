import React from 'react';
import SearchResults from './SearchResults.js';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import FilterButton from '../FilterButton/FilterButton.js';

describe('SearchResults component', () => {

    let count;

    beforeAll(() => {
        count='';
    });

    it('should be render correctly', () => {
        const component = shallow(<SearchResults count={count} />);
        
        expect(component).toMatchSnapshot();
    });

    it('test filter rating button click event',() => {
        const component = shallow(<SearchResults  count={count} />);
        component.instance().handleClick = jest.fn();

        component.update();
        component.find('#rating').simulate('click', { target: { id: ''}});
        expect(component.instance().handleClick).toHaveBeenCalled;
    });

    it('test filter release-date button click event',() => {
        const component = shallow(<SearchResults  count={count} />);
        component.instance().handleClick = jest.fn();

        component.update();
        component.find('#release-date').simulate('click', { target: { id: ''}});
        expect(component.instance().handleClick).toHaveBeenCalled;
    });
});
