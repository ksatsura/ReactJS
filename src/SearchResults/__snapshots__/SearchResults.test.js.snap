// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component should be render correctly 1`] = `ShallowWrapper {}`;
